tests/Integration/Behaviour/Features/Scenario/CartRule/set_cart_rule_restrictions.feature

```diff
@@ -0,0 +1,111 @@
+      | name[en-US]                            | Half the price      |
+      | is_active                              | true                |
+      | allow_partial_use                      | true                |
+      | priority                               | 3                   |
+      | valid_from                             | 2022-01-01 11:00:00 |
+      | valid_to                               | 3001-01-01 12:00:00 |
+      | total_quantity                         | 10                  |
+      | quantity_per_user                      | 12                  |
+      | free_shipping                          | false               |
+      | code                                   | rule_70_percent     |
+      | reduction_percentage                   | 70                  |
+      | reduction_apply_to_discounted_products | false               |
+    And cart rule "rule_free_shipping_1" should have the following properties:
+      | is_active             | true                 |
+      | free_shipping         | true                 |
+      | priority              | 1                    |
+      | valid_from            | 2022-01-01 11:00:00  |
+      | valid_to              | 3001-01-01 12:00:00  |
+      | code                  | rule_free_shipping_1 |
+      | restricted cart rules |                      |
+    And cart rule "rule_50_percent" should have the following properties:
+      | is_active                              | true                   |
+      | priority                               | 2                      |
+      | valid_from                             | 2022-01-01 11:00:00    |
+      | valid_to                               | 3001-01-01 12:00:00    |
+      | free_shipping                          | false                  |
+      | code                                   | rule_50_percent        |
+      | reduction_percentage                   | 50                     |
+      | reduction_apply_to_discounted_products | false                  |
+      | discount_application_type              | order_without_shipping |
+      | restricted cart rules                  |                        |
+    And cart rule "rule_70_percent" should have the following properties:
+      | is_active                              | true                   |
+      | priority                               | 3                      |
+      | valid_from                             | 2022-01-01 11:00:00    |
+      | valid_to                               | 3001-01-01 12:00:00    |
+      | free_shipping                          | false                  |
+      | code                                   | rule_70_percent        |
+      | reduction_percentage                   | 70                     |
+      | reduction_apply_to_discounted_products | false                  |
+      | discount_application_type              | order_without_shipping |
+      | restricted cart rules                  |                        |
```

Comment:
Didn't add dedicated step for `restricted cart rules` check because they are all inside CartRuleForEditing, so the class is asserted in one step. And the properties to assert are optional so you are not forced to always assert full class. (though thisleaves us the chance to forget to forget asserting some property, but requiring to assert ALL the properties all the time would be a pain and would make the tests unreadable :sweat: )